cloudant-sync-datastore-android/src/main/java/com/cloudant/sync/replication/ReplicationService.java

```diff
@@ -153,18 +157,25 @@ public void onCreate() {
         // Get the HandlerThread's Looper and use it for our Handler.
         Looper serviceLooper = thread.getLooper();
         mServiceHandler = getHandler(serviceLooper);
+    }
 
-        Replicator[] replicators = getReplicators(getApplicationContext());
-
+    public void setReplicators(Replicator[] replicators) {
```

Comment:
Fixed in 7f971d3. Also added some additional tests and some javadoc to the `setReplicators()` and changed the existing `RuntimeException` to a more specific `IllegalArgumentException`.